### PR TITLE
GSK-2545 Update `Suite.upload()` method to upload modifications to existing suite

### PR DESCRIPTION
**Pull request by kevinmessiaen**
## Description

Updated `Suite.upload()` method to upload modifications to existing suite

### Add test to a Suite:
```python
import giskard
client = giskard.GiskardClient(
    url="http://localhost:9000",  # URL of your Giskard instance
    key="gsk-...")

suite = giskard.Suite.download(client, 'test', 52)

from giskard.testing.tests.data_quality import test_data_uniqueness

suite.add_test(test_data_uniqueness())
suite.upload(client, 'test')
```

### Edit a test from a Suite:
```python
import giskard
client = giskard.GiskardClient(
    url="http://localhost:9000",  # URL of your Giskard instance
    key="gsk-...")

suite = giskard.Suite.download(client, 'test', 52)

suite.update_test_params(len(suite.tests) - 1, threshold=0.2) # Not ideal by index but will be changed in another PR
suite.upload(client, 'test')
```

### Remove test from a Suite:
```python
import giskard
client = giskard.GiskardClient(
    url="http://localhost:9000",  # URL of your Giskard instance
    key="gsk-...")

suite = giskard.Suite.download(client, 'test', 52)

from giskard.testing.tests.data_quality import test_data_uniqueness

suite.remove_test(test_data_uniqueness())
suite.upload(client, 'test')
```

## Type of Change

- [ ] 📚 Examples / docs / tutorials / dependencies update
- [ ] 🔧 Bug fix (non-breaking change which fixes an issue)
- [ ] 🥂 Improvement (non-breaking change which improves an existing feature)
- [x] 🚀 New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🔐 Security fix

## Checklist

- [ ] I've read the [`CODE_OF_CONDUCT.md`](https://github.com/Giskard-AI/ai-inspector/blob/master/CODE_OF_CONDUCT.md) document.
- [ ] I've read the [`CONTRIBUTING.md`](https://github.com/Giskard-AI/ai-inspector/blob/master/CONTRIBUTING.md) guide.
- [ ] I've updated the code style using `make codestyle`.
- [ ] I've written tests for all new methods and classes that I created.
- [ ] I've written the docstring in Google format for all the methods and classes that I used.

### CODE REVIEW

File: giskard/core/suite.py [hunk 1]
```diff
@@ -435,10 +441,15 @@ def upload(self, client: GiskardClient, project_key: str):
...
-        self.id = client.save_test_suite(self.to_dto(client, project_key, uploaded_uuid_status))
+        if self.id:
+            client.update_test_suite(self.id, self.to_dto(client, project_key, uploaded_uuid_status))
+            analytics.track("hub:test_suite:updated")
+        else:
+            self.id = client.save_test_suite(self.to_dto(client, project_key, uploaded_uuid_status))
+            analytics.track("hub:test_suite:uploaded")
 
         print(f"Test suite has been saved: {client.host_url}/main/projects/{project_key}/test-suite/{self.id}/overview")
-        analytics.track("hub:test_suite:uploaded")
```
Comment (Inokinoki) [inline giskard/core/suite.py:441]: Should we track in both operations: creation by uploading and update by uploading?

File: giskard/ml_worker/websocket/listener.py [hunk 2]
```diff
@@ -563,7 +563,7 @@ def run_test_suite(
                 test_args: Dict[str, Any] = copy(test_args)
                 test_args.update(**test_args.pop("kwargs"))
             updated_test_args.append(test_args)
-            suite.add_test(t["test"].get_builder()(**test_args), t["id"])
```
Comment (Inokinoki) [inline giskard/ml_worker/websocket/listener.py:566]: Why aren't we using builder now?

File: giskard/core/suite.py [hunk 3]
```diff
@@ -66,12 +66,12 @@ def parse_function_arguments(client, project_key, function_inputs):
             arguments[value["name"]] = str(value["value"])
         elif value["type"] == "bool":
             arguments[value["name"]] = bool(value["value"])
-        elif value["type"] == "kwargs":
+        elif value["type"] == "Kwargs":
```
Comment (Inokinoki) [inline giskard/core/suite.py:69]: Is this a typo now or was it a typo before?

File: giskard/core/suite.py (hunk 1 above)
Comment (kevinmessiaen) [reply giskard/core/suite.py:479]: Yes it has been moved to line 449

File: giskard/ml_worker/websocket/listener.py (hunk 2 above)
Comment (kevinmessiaen) [reply giskard/ml_worker/websocket/listener.py:566]: The `__call__` method now always makes a copy of the object using `instance = copy.deepcopy(self)` so we don't need the `get_builder` method anymore.

File: giskard/core/suite.py (hunk 3 above)
Comment (kevinmessiaen) [reply giskard/core/suite.py:69]: It was before, `Kwargs` are uppercase in the Hub.

**Review by andreybavt — APPROVED** [review]

**Pull request merged**